pkg/ruler/appender.go

```diff
@@ -0,0 +1,183 @@
+package ruler
+
+import (
+	"context"
+	"errors"
+
+	"github.com/grafana/loki/pkg/util"
+
+	"github.com/prometheus/prometheus/rules"
+
+	"github.com/prometheus/prometheus/promql"
+
+	"github.com/prometheus/prometheus/pkg/exemplar"
+
+	"github.com/cortexproject/cortex/pkg/cortexpb"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/storage"
+)
+
+type RemoteWriteAppendable struct {
+	groupAppender map[string]*RemoteWriteAppender
+
+	userID    string
+	cfg       Config
+	overrides RulesLimits
+	logger    log.Logger
+}
+
+type RemoteWriteAppender struct {
+	logger       log.Logger
+	ctx          context.Context
+	remoteWriter remoteWriter
+	userID       string
+	groupKey     string
+
+	queue *util.EvictingQueue
+}
+
+func (a *RemoteWriteAppendable) Appender(ctx context.Context) storage.Appender {
+	if err := a.cfg.RemoteWrite.Validate(); err != nil {
```

Comment:
Do we need to call this again here if we already do it in `newAppendable` and return a `NoopAppender` if it errors?
Also, we do not reload the config other than overrides and runtime config so we should not expect it to change.

Reply:
Correct - this will get refactored out based on your and Cyril's feedback - thanks!